devtools/create_project/cmake.cpp

```diff
@@ -243,38 +281,21 @@ void CMakeProvider::createProjectFile(const std::string &name, const std::string
 	if (!project)
 		error("Could not open \"" + projectFile + "\" for writing");
 
-	bool addEnableCheck = true;
 	if (name == setup.projectName) {
 		project << "add_executable(" << name << "\n";
-		addEnableCheck = false;
 	} else if (name == setup.projectName + "-detection") {
 		project << "list(APPEND SCUMMVM_LIBS " << name << ")\n";
 		project << "add_library(" << name << "\n";
-		addEnableCheck = false;
 	} else {
-		std::string engineName;
-		std::transform(name.begin(), name.end(), std::back_inserter(engineName), toupper);
-
-		project << "if (ENABLE_" << engineName << ")\n";
-		project << "add_library(" << name << "\n";
-	}
-
-	std::string modulePath;
-	if (!moduleDir.compare(0, setup.srcDir.size(), setup.srcDir)) {
-		modulePath = moduleDir.substr(setup.srcDir.size());
-		if (!modulePath.empty() && modulePath.at(0) == '/')
-			modulePath.erase(0, 1);
+		enginesStr << "add_engine(" << name << "\n";
```

Comment:
I'm using a buffer because the engines have to come after the main project. In a follow-up commit, I write the engines to separate files, and this buffer won't be used (when running in-source).

Reply:
I can come up with alternatives if you're worried about memory consumption (I suppose it should be several 100k for a full build), or anything else.

Reply:
create_project is a development tool used in desktop PCs, which have more than enough memory. Some extra kilobytes of RAM won't be an issue in this case